# Kansen

## Laplace
Franse wiskundige en astronoom

### Kansen volgens Laplace
De kans dat een gebeurtenis A gebeurt is de verhouding van het aantal gunstige uitkomsten voor A tot het aantal mogelijke uitkomsten.
$$P(A) = \frac{aantal\ gunstige\ uitkomsten\ voor\ A}{aantal\ mogelijke\ uitkomsten}$$
Voorbeeld:
- Een rode knikker trekken uit een pot met 30 rode en 20 blauwe knikkers.
- De kans op een rode knikker is $\frac{30}{50}⇔0.6$.

Een kans is altijd een getal tussen 0 en 1.
- 0: onmogelijk
- 0.5: even waarschijnlijk als onwaarschijnlijk 
- 1: zeker
- Als je een getal kleiner dan 0 of groter dan 1 krijgt, heb je een fout gemaakt.

De formule kan ook zichtbaar gemaakt worden.
<br>
<img src="rescources/img/Schermafbeelding 2024-03-17 091947.png">

- De verzameling van alle mogelijke uitkomsten noemen we U (universum).
- De verzameling van alle gunstige uitkomsten noemen we G.
- De kans dat G gebeurt is gelijk aan $P(G)=\frac{#G}{#U}$

### Relatieve frequentie
Een kans kan ook bekeken worden als een relatieve frequentie.
- Stel dat je 1000 keer een knikker uit de zak neemt (en weer terug zou leggen), dan zou je waarschijnlijk 600 keer een rode knikker hebben.
- Hoe meer keer je een knikker uit de zak neemt, hoe dichter de relatieve frequentie bij de kans komt.

In Python is dit te simuleren met:

In [25]:
import random

#Maak een zak met 20 blauwe en 30 rode knikkers
zak = ['blauw'] * 20 + ['rood'] * 30
#Zet de random seed op 5 (uitkomst zal altijd hetzelfde zijn)
random.seed(5)
#Neem 10 knikkers uit de zak
steekproef = random.choices(zak, k=10)
print(steekproef)

['rood', 'rood', 'rood', 'rood', 'rood', 'rood', 'blauw', 'rood', 'rood', 'rood']


In [26]:
import pandas as pd

pd.Series(steekproef).value_counts(normalize=True)

rood     0.9
blauw    0.1
Name: proportion, dtype: float64

Bij deze steekproef was de kans op een rode knikker 90%.
Dit is ver van de berkekende kan van 60%.
Dit komt omdat de steekproef te klein is. We hebben maar 10 knikkers getrokken.
Als we de steekproef groter maken, zal de relatieve frequentie dichter bij de kans komen: 

In [27]:
random.seed(5)
steekproef = random.choices(zak, k=100)
pd.Series(steekproef).value_counts(normalize=True)

rood     0.54
blauw    0.46
Name: proportion, dtype: float64

In [28]:
random.seed(5)
steekproef = random.choices(zak, k=1000)
pd.Series(steekproef).value_counts(normalize=True)

rood     0.581
blauw    0.419
Name: proportion, dtype: float64

Als de steekproef oneindig groot zou zijn, zou de relatieve frequentie gelijk zijn aan de kans.
Een kans kunnen we dus definiëren als de limiet van de relatieve frequentie als de steekproefgrootte naar oneindig gaat.

## Rekenen met kansen

Als je de kans van een gebeurtenis wil vinden, dan kan je dat soms doen aan de hand van de kans van andere (deel)gebeurtenissen. We zien in eerste instantie 3 veel voorkomende formules: de inverse, de productregel en de somregel.

### De Inverse

Als je de kans weet van een gebeurtenis, dan kan je heel gemakkelijk de kans berekenen dat deze gebeurtenis niet zal gebeuren.
$P(G̅) = 1 - P(G)$

### De productregel
Bereken de kans dat 2 gebeurtenissen tegelijk optreden.

#### De algemene productregel
Als je de kans wil berekenen dat twee gebeurtenissen A en B gebeuren, dan kan je dat doen aan de hand van de kans van A en de kans van B.
Om dit duidelijk te maken gebruiken we een voorbeeld:
Stel:
- Een groep kinderen bestaat voor 40% uit meisjes en 60% uit jongens.
- 10% van de kinderen draagt een bril.

Als we een willekeurig kind uit de groep nemen is de kans dat het een meisje is 40% en de kans dat het een bril draagt 10%.
- $P(meisje) = 0.4$ en $P(bril) = 0.1$

Als we nu willen weten wat de kans is dat het kind een meisje EN een bril draagt missen we nog één gegeven.
We moeten weten wat de kans is dat een meisje een bril draagt. Deze kunnen we noteren als $P(bril|meisje)$.
De kansen zijn afhankelijk van elkaar.

Als we de kans van $P(bril|meisje)$ weten, dan kunnen we de kans van $P(meisje\ en\ bril)$ berekenen met de volgende formule:
$$P(meisje\ en\ bril) = P(meisje) * P(bril|meisje)$$

Stel dat de kans dat een meisje een bril draagt 20% is,
Dan kunnen we de formule invullen met de gegevens:
$$P(meisje\ en\ bril) = 0.4 * 0.2 = 0.08 ⇔ 8\%$$
<img src="rescources/img/Schermafbeelding 2024-03-17 100130.png">
<br>
De algemene productregel is:
$$P(A\ en\ B) = P(A) * P(B|A)$$

#### Onafhankelijke gebeurtenissen
In sommige gevallen zijn de kansen van twee gebeurtenissen niet afhankelijk van elkaar. Dit noemen we onafhankelijke gebeurtenissen.
Een voorbeeld hiervan:
Wat is de kans dat iemand een rood tshirt draagt en goed is in wiskunde?
$$P(rood\ EN\ wiskunde) = P(rood) * P(wiskunde|rood)$$
Maar omdat de kans dat iemand goed is in wiskunde niet afhankelijk is van de kleur van het tshirt, kunnen we de formule ook schrijven als:
$$P(rood\ EN\ wiskunde) = P(rood) * P(wiskunde)$$

### De somregel
Bereken de kans dat één van twee gebeurtenissen optreedt of allebei.

Deze kans kan worden berekend worden met volgende formule:
$$P(A\ of\ B) = P(A) + P(B) - P(A\ en\ B)$$

We kunnen deze formule toepassen op het vorige voorbeeld:
$P(meisje\ of\ bril) = P(meisje) + P(bril) - P(meisje\ en\ bril)$
<br>
<img src="rescources/img/Schermafbeelding 2024-03-17 103055.png">
Het is zeer belangrijk om de kans dat de 2 gebeurtenissen samen gebeuren terug af te trekken van de som van de kansen van de 2 gebeurtenissen. Als we dit niet doen, dan tellen we de kans dat de 2 gebeurtenissen samen gebeuren 2 keer. Dit kan je zien aan de afbeelding hierboven. Als we de kans niet terug aftrekken dan zou het rood gearceerde deel 2 keer geteld worden.

Als we de data van het voorbeeld invullen krijgen we:
$$P(meisje\ of\ bril) = 0.4 + 0.1 - 0.08 = 0.42 ⇔ 42\%$$

## De regel van Bayes
De productregel zegt: $P(A\ en\ B) = P(A) * P(B|A)$
Maar volgende formule is ook waar: $P(A\ en\ B) = P(B) * P(A|B)$
Het is mogelijk om A en B om te wisselen. Dit betekent dat:
$$P(A) * P(B|A) = P(B) * P(A|B)$$
hieruit volgt:
$$P(B|A) = \frac{P(A|B) * P(B)}{P(A)}$$
of
$$P(A|B) = \frac{P(B|A) * P(A)}{P(B)}$$

Hierdoor kunnnen we $P(A|B)$ berekenen aan de hand van $P(B|A)$ en vice versa.

## Wet van de totale kans
De wet van de totale kans is niet echt een nieuwe formule maar eerder een samenstelling van de vorige formules.
De wet van de totale kans kan gebruikt worden om de totale kans te berekenen van een gebeurtenis A waarbij er een aantal deelgebeurtenissen zijn die nooit samen kunnen voorkomen.

$$P(A) = P(G_1) * P(A|G_1) + P(G_2) * P(A|G_2) + ... + P(G_n) * P(A|G_n)$$